test/e2e/set/scyllacluster/scyllamanager.go

```diff
@@ -85,20 +115,32 @@ var _ = g.Describe("Scylla Manager integration", func() {
 			return false, nil
 		}
 
-		waitCtx2, waitCtx2Cancel := utils.ContextForManagerSync(ctx, sc)
-		defer waitCtx2Cancel()
-		sc, err = controllerhelpers.WaitForScyllaClusterState(waitCtx2, f.ScyllaClient().ScyllaV1().ScyllaClusters(sc.Namespace), sc.Name, controllerhelpers.WaitForStateOptions{}, registeredInManagerCond, repairTaskScheduledCond, backupTaskSyncFailedCond)
+		waitCtx3, waitCtx3Cancel := utils.ContextForManagerSync(ctx, sc)
+		defer waitCtx3Cancel()
+		sc, err = controllerhelpers.WaitForScyllaClusterState(waitCtx3, f.ScyllaClient().ScyllaV1().ScyllaClusters(sc.Namespace), sc.Name, controllerhelpers.WaitForStateOptions{}, repairTaskScheduledCond, backupTaskSchedulingFailedCond)
 		o.Expect(err).NotTo(o.HaveOccurred())
 
 		managerClient, err := utils.GetManagerClient(ctx, f.KubeAdminClient().CoreV1())
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		framework.By("Verifying that task properties were synchronized")
+		framework.By("Verifying that repair task properties were synchronized")
 		tasks, err := managerClient.ListTasks(ctx, *sc.Status.ManagerID, "repair", false, "", "")
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(tasks.TaskListItemSlice).To(o.HaveLen(1))
 		repairTask := tasks.TaskListItemSlice[0]
-		o.Expect(repairTask.Name).To(o.Equal(sc.Spec.Repairs[0].Name))
+		o.Expect(repairTask.Name).To(o.Equal(sc.Status.Repairs[0].Name))
+		o.Expect(repairTask.ID).To(o.Equal(sc.Status.Repairs[0].ID))
 		o.Expect(repairTask.Properties.(map[string]interface{})["parallel"].(json.Number).Int64()).To(o.Equal(int64(123)))
+
+		framework.By("Waiting for repair to finish")
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 10*time.Minute, true, func(context.Context) (done bool, err error) {
```

Comment:
should this be propagated to scyllacluster status? if yes and we miss the information, a todo with an issue link seems to be in order.

Reply:
to my knowledge we don't sync the tasks post-scheduling, so it's not that we miss the status propagation, we don't have the logic to handle the progress. I'm not sure if that could work with recurring tasks though. @zimnx mind sharing your thoughts on if that's viable?

Reply:
yeah, with recurring tasks we'd have to parse it and schedule queue add at that point - probably not what we want to get into - I take this back :)